diff --git a/modules/lwjgl/core/src/generated/c/linux/org_lwjgl_system_linux_UIO.c b/modules/lwjgl/core/src/generated/c/linux/org_lwjgl_system_linux_UIO.c
index f328ca6..5c29d71 100644
--- a/modules/lwjgl/core/src/generated/c/linux/org_lwjgl_system_linux_UIO.c
+++ b/modules/lwjgl/core/src/generated/c/linux/org_lwjgl_system_linux_UIO.c
@@ -5,6 +5,7 @@
  */
 #include "common_tools.h"
 #include <sys/uio.h>
+#include "droid_uio.h"
 #include <errno.h>
 
 EXTERN_C_ENTER
diff --git a/modules/lwjgl/core/src/templates/kotlin/core/linux/templates/uio.kt b/modules/lwjgl/core/src/templates/kotlin/core/linux/templates/uio.kt
index 87d339f..19b448d 100644
--- a/modules/lwjgl/core/src/templates/kotlin/core/linux/templates/uio.kt
+++ b/modules/lwjgl/core/src/templates/kotlin/core/linux/templates/uio.kt
@@ -10,6 +10,7 @@ import org.lwjgl.generator.*
 val uio = "UIO".nativeClass(Module.CORE_LINUX, nativeSubPath = "linux") {
     nativeImport(
         "<sys/uio.h>",
+        "droid_uio.h",
         "<errno.h>"
     )
     // TODO:
@@ -108,4 +109,4 @@ val uio = "UIO".nativeClass(Module.CORE_LINUX, nativeSubPath = "linux") {
         unsigned_long_int("__riovcnt"),
         unsigned_long_int("__flags")
     )
-}
\ No newline at end of file
+}
diff --git a/modules/lwjgl/core/src/main/c/droid_uio.h b/modules/lwjgl/core/src/main/c/droid_uio.h
new file mode 100644
index 0000000..59a2c8a
--- /dev/null
+++ b/modules/lwjgl/core/src/main/c/droid_uio.h
@@ -0,0 +1,32 @@
+#include <unistd.h>
+#include <errno.h>
+#include <sys/uio.h>
+#include <sys/syscall.h>
+
+inline ssize_t process_vm_readv (pid_t pid, const struct iovec *local_iov,
+		  unsigned long int liovcnt,
+		  const struct iovec *remote_iov,
+		  unsigned long int riovcnt, unsigned long int flags) {
+  return syscall(__NR_process_vm_readv, pid, local_iov,
+			      liovcnt, remote_iov, riovcnt, flags);
+}
+
+
+inline ssize_t process_vm_writev (pid_t pid, const struct iovec *local_iov,
+		   unsigned long int liovcnt,
+		   const struct iovec *remote_iov,
+		   unsigned long int riovcnt, unsigned long int flags) {
+  return syscall(__NR_process_vm_writev, pid, local_iov,
+			      liovcnt, remote_iov, riovcnt, flags);
+}
+
+inline ssize_t preadv(int fd, const struct iovec *iov, int iovcnt,
+                       off_t offset) {
+  return syscall(__NR_preadv, fd, iov, iovcnt, offset);
+}
+
+
+inline ssize_t pwritev(int fd, const struct iovec *iov, int iovcnt,
+                       off_t offset) {
+  return syscall(__NR_pwritev, fd, iov, iovcnt, offset);
+}
diff --git a/config/linux/build.xml b/config/linux/build.xml
index 7108ff7..ac7a388 100644
--- a/config/linux/build.xml
+++ b/config/linux/build.xml
@@ -45,7 +45,7 @@
                 <arg line="-O3 -flto=auto -fPIC @{flags} -pthread -DNDEBUG -DLWJGL_LINUX -DLWJGL_${build.arch}"/>
                 <arg line="-U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=0 -D_GNU_SOURCE"/>
                 <arg line="-D_FILE_OFFSET_BITS=64"/>
-
+		<arg line="${env.CFLAGS}" if:set="env.CFLAGS"/>
                 <arg value="-I${jni.headers}"/>
                 <arg value="-I${jni.headers}/linux"/>
 
@@ -480,4 +480,4 @@
             <touch file="${lib.native}/touch.txt" verbose="false"/>
         </sequential>
     </target>
-</project>
\ No newline at end of file
+</project>
