diff --git a/modules/lwjgl/core/src/templates/kotlin/core/linux/templates/uio.kt b/modules/lwjgl/core/src/templates/kotlin/core/linux/templates/uio.kt
index 87d339f..19b448d 100644
--- a/modules/lwjgl/core/src/templates/kotlin/core/linux/templates/uio.kt
+++ b/modules/lwjgl/core/src/templates/kotlin/core/linux/templates/uio.kt
@@ -10,4 +10,5 @@ import org.lwjgl.generator.*
     nativeImport(
         "<sys/uio.h>",
+        "droid_uio.h",
         "<errno.h>"
     )
diff --git a/modules/lwjgl/core/src/main/c/droid_uio.h b/modules/lwjgl/core/src/main/c/droid_uio.h
new file mode 100644
index 0000000..783d34f
--- /dev/null
+++ b/modules/lwjgl/core/src/main/c/droid_uio.h
@@ -0,0 +1,44 @@
+#include <unistd.h>
+#include <errno.h>
+#include <sys/uio.h>
+#include <sys/syscall.h>
+
+inline ssize_t process_vm_readv (pid_t pid, const struct iovec *local_iov,
+		  unsigned long int liovcnt,
+		  const struct iovec *remote_iov,
+		  unsigned long int riovcnt, unsigned long int flags) {
+  return syscall(__NR_process_vm_readv, pid, local_iov,
+			      liovcnt, remote_iov, riovcnt, flags);
+}
+
+
+inline ssize_t process_vm_writev (pid_t pid, const struct iovec *local_iov,
+		   unsigned long int liovcnt,
+		   const struct iovec *remote_iov,
+		   unsigned long int riovcnt, unsigned long int flags) {
+  return syscall(__NR_process_vm_writev, pid, local_iov,
+			      liovcnt, remote_iov, riovcnt, flags);
+}
+
+inline ssize_t preadv(int fd, const struct iovec *iov, int iovcnt,
+                       off_t offset) {
+  return syscall(__NR_preadv, fd, iov, iovcnt, offset);
+}
+
+
+inline ssize_t pwritev(int fd, const struct iovec *iov, int iovcnt,
+                       off_t offset) {
+  return syscall(__NR_pwritev, fd, iov, iovcnt, offset);
+}
+
+// TODO: wrap these correctly
+inline ssize_t preadv2(int fd, const struct iovec *iov, int iovcnt,
+                       off_t offset, int flags) {
+  errno = ENOSYS;
+  return -1;
+}
+inline ssize_t pwritev2(int fd, const struct iovec *iov, int iovcnt,
+                       off_t offset, int flags) {
+  errno = ENOSYS;
+  return -1;
+}