diff --git a/modules/lwjgl/core/src/main/java/org/lwjgl/system/MemoryUtil.java b/modules/lwjgl/core/src/main/java/org/lwjgl/system/MemoryUtil.java
index 2281bfd..0ecbaac 100644
--- a/modules/lwjgl/core/src/main/java/org/lwjgl/system/MemoryUtil.java
+++ b/modules/lwjgl/core/src/main/java/org/lwjgl/system/MemoryUtil.java
@@ -61,6 +61,8 @@ public final class MemoryUtil {
     static final ThreadLocal<char[]> ARRAY_TLC_CHAR = ThreadLocal.withInitial(() -> new char[ARRAY_TLC_SIZE]);
 
     static final MemoryBackend BACKEND;
+    // Float accesses on ARM require 4-byte (or 8-byte) alignment, so we need to make some special considerations for this arch to avoid exploding Java.
+    private static final boolean FLOAT_REQUIRES_ALIGNMENT = "arm".equals(System.getProperty("os.arch"));
 
     static final Charset UTF16 = ByteOrder.nativeOrder() == ByteOrder.LITTLE_ENDIAN
         ? StandardCharsets.UTF_16LE
@@ -2594,10 +2596,10 @@ public final class MemoryUtil {
     public static long memGetLong(long ptr)                          { return BACKEND.getLong(ptr); }
     public static long memGetLongUnaligned(long ptr)                 { return BACKEND.getLongUnaligned(ptr); }
 
-    public static float memGetFloat(long ptr)                        { return BACKEND.getFloat(ptr); }
+    public static float memGetFloat(long ptr)                        { return FLOAT_REQUIRES_ALIGNMENT ? Float.intBitsToFloat(BACKEND.getInt(ptr)) : BACKEND.getFloat(ptr); }
     public static float memGetFloatUnaligned(long ptr)               { return BACKEND.getFloatUnaligned(ptr); }
 
-    public static double memGetDouble(long ptr)                      { return BACKEND.getDouble(ptr); }
+    public static double memGetDouble(long ptr)                      { return FLOAT_REQUIRES_ALIGNMENT ? Double.longBitsToDouble(BACKEND.getInt(ptr)) : BACKEND.getDouble(ptr); }
     public static double memGetDoubleUnaligned(long ptr)             { return BACKEND.getDoubleUnaligned(ptr); }
 
     public static long memGetCLong(long ptr)                         { return BACKEND.getCLong(ptr); }
@@ -2622,10 +2624,10 @@ public final class MemoryUtil {
     public static void memPutLong(long ptr, long value)              { BACKEND.putLong(ptr, value); }
     public static void memPutLongUnaligned(long ptr, long value)     { BACKEND.putLongUnaligned(ptr, value); }
 
-    public static void memPutFloat(long ptr, float value)            { BACKEND.putFloat(ptr, value); }
+    public static void memPutFloat(long ptr, float value)            { if(FLOAT_REQUIRES_ALIGNMENT) BACKEND.putInt(ptr, Float.floatToRawIntBits(value)); else BACKEND.putFloat(ptr, value); }
     public static void memPutFloatUnaligned(long ptr, float value)   { BACKEND.putFloatUnaligned(ptr, value); }
 
-    public static void memPutDouble(long ptr, double value)          { BACKEND.putDouble(ptr, value); }
+    public static void memPutDouble(long ptr, double value)          { if(FLOAT_REQUIRES_ALIGNMENT) BACKEND.putLong(ptr, Double.doubleToRawLongBits(value)); else BACKEND.putDouble(ptr, value); }
     public static void memPutDoubleUnaligned(long ptr, double value) { BACKEND.putDoubleUnaligned(ptr, value); }
 
     public static void memPutCLong(long ptr, long value)             { BACKEND.putCLong(ptr, value); }
