diff --git a/modules/lwjgl/rpmalloc/src/main/c/rpmalloc.c b/modules/lwjgl/rpmalloc/src/main/c/rpmalloc.c
index f0412ce..a8698eb 100644
--- a/modules/lwjgl/rpmalloc/src/main/c/rpmalloc.c
+++ b/modules/lwjgl/rpmalloc/src/main/c/rpmalloc.c
@@ -2129,7 +2129,11 @@ _memory_unmap_os(void* address, size_t size, size_t offset, size_t release) {
 #if defined(POSIX_MADV_FREE)
 		if (posix_madvise(address, size, POSIX_MADV_FREE))
 #endif
+#if defined(MADV_DONTNEED)
+		if (madvise(address, size, MADV_DONTNEED)) {
+#else
 		if (posix_madvise(address, size, POSIX_MADV_DONTNEED)) {
+#endif
 			assert("Failed to madvise virtual memory block as free" == 0);
 		}
 	}
