diff --git a/modules/lwjgl/core/src/main/c/linux/LinuxLWJGL.h b/modules/lwjgl/core/src/main/c/linux/LinuxLWJGL.h
index 4fe2e13..e69de29 100644
--- a/modules/lwjgl/core/src/main/c/linux/LinuxLWJGL.h
+++ b/modules/lwjgl/core/src/main/c/linux/LinuxLWJGL.h
@@ -1,8 +0,0 @@
-/*
- * Copyright LWJGL. All rights reserved.
- * License terms: https://www.lwjgl.org/license
- */
-#pragma once
-
-#include <X11/X.h>
-#include <X11/Xlib.h>
